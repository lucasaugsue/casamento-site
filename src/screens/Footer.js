import Menu from '@mui/icons-material/Menu';
import { Collapse, List, ListItemButton, ListItemText } from '@mui/material';
import React from 'react';
import styles from './Footer.module.css';
import InstagramIcon from '@mui/icons-material/Instagram';
import LinkedInIcon from '@mui/icons-material/LinkedIn';
import GitHubIcon from '@mui/icons-material/GitHub';
import GoogleIcon from '@mui/icons-material/Google';
import CallIcon from '@mui/icons-material/Call';
import EmailIcon from '@mui/icons-material/Email';
import ClientContext from '../contexts/ClientContext';

export default function Footer(){
    const { apiRequest } = React.useContext(ClientContext);

    const listItens = [
        {id: 3, title: "Confirme sua presença", section: "#confirmar-presenca"},
        {id: 4, title: "Escreva um recado", section: "#recado"},
        {id: 5, title: "Lista de presentes", section: "#presentes"},
    ]

    // get presentes
    const [presentes, setPresentes] = React.useState([]);
    const [loading, setLoading] = React.useState(false);

    const getPresentes = () => {
        setLoading(true)
        apiRequest("GET", "/presentes/list")
        .then((res) => {
            setPresentes(res)
            setLoading(false)
        })
        .catch((err) => {
            setLoading(false)
            showNotification({message: err.message, color: 'red', autoClose: true})
        });
    }

    React.useEffect(() => {
        getPresentes()
    }, []);

    return <div>
        <div className={styles.redesSociaisContainer}>
            <div className={styles.redesSociaisText}> 
                Siga nas redes sociais
            </div>
            <div/>
            <div className={styles.containerIcons}>
                <div className={styles.iconsRedesSociais}>
                    <InstagramIcon onClick={() => window.open("https://www.instagram.com/lucas_augsue/")}/>
                </div> 
                <div className={styles.iconsRedesSociais}>
                    <LinkedInIcon onClick={() => window.open("https://www.linkedin.com/in/lucas-augsue/")}/>
                </div> 
                <div className={styles.iconsRedesSociais}> 
                    <GitHubIcon onClick={() => window.open("https://github.com/lucasaugsue")}/>
                </div>
                <div className={styles.iconsRedesSociais}>
                    <GoogleIcon onClick={() => window.open("https://www.google.com/search?q=lucas+augsue&client=opera&hs=UyF&sca_esv=591504270&sxsrf=AM9HkKko2i1mK_ahxmXKk2mwl9tJm_rLaw%3A1702739299042&ei=Y719Za-OAvW35OUP65GAmAo&ved=0ahUKEwjvy8GYnpSDAxX1G7kGHesIAKMQ4dUDCBA&uact=5&oq=lucas+augsue&gs_lp=Egxnd3Mtd2l6LXNlcnAiDGx1Y2FzIGF1Z3N1ZUiKB1CxBFjHBXABeACQAQCYAf4BoAH-AaoBAzItMbgBA8gBAPgBAeIDBBgBIEGIBgE&sclient=gws-wiz-serp#ip=1")}/>
                </div>
            </div>
        </div>

        <div className={styles.maisInformacoesContainer}>
            <div className={styles.gridInformacoes}> 

                <div className={styles.maisInformacoesTitle}> 
                    Mais informações 
                </div>

                <div className={styles.maisInformacoesBody}> 
                    Para auxiliar no meu sonho em casar no meio do ano de 2024, por isso resolvi eu 
                    mesmo fazer o site do meu casamento 
                </div>

            </div>
            <div className={styles.gridInformacoes}> 

                <div className={styles.maisInformacoesTitle}> 
                    Presentes
                </div>

                {!loading && [...presentes.slice(0,3)]
                    .map((item, index) => <div 
                        key={`${index};;${item.id}`} 
                        className={styles.textLink}
                        onClick={() => window.location.href = `/info/${item.id}`}
                    > 
                        {
                            (item.nome && item.nome.length > 32)
                            ? `${item.nome.slice(0, 30)}...`
                            : item.nome || ""
                        }
                    </div>
                )}

            </div>
            <div className={styles.gridInformacoes}> 

                <div className={styles.maisInformacoesTitle}> 
                    Links importantes 
                </div>

                {listItens.map((i, index) => 
                    <div className={styles.textLink}
                        key={`${i.id};;${index}`}
                        onClick={() => {
                            window.location.href=`/${i.section}`
                        }}
                    > {i.title} </div>
                )}
            </div>
            <div className={styles.gridInformacoes}> 

                <div className={styles.maisInformacoesTitle}> 
                    Contato
                </div>

                <div className={styles.boxContato}>
                    <EmailIcon/> 
                    <div className={styles.maisInformacoesBody}> 
                        lucasaugsue7@gmail.com
                    </div>
                </div>

                <div className={styles.boxContato}>
                    <CallIcon/> 
                    <div className={styles.maisInformacoesBody}> 
                        (61)98114-6060
                    </div>
                </div>

            </div>
        </div>

        <div className={styles.copyrightContainer}>
            <div/>
            <div className={styles.copyrightText}> 
                @ 2023 Copyright lucasaugsue 
            </div>
            <div/>
        </div> 
    </div>
}
