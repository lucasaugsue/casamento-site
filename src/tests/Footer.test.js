import '@testing-library/jest-dom';
import renderer, { act } from 'react-test-renderer';
import ClientContext from '../contexts/ClientContext';
import Footer from '../screens/Footer';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';

/**
 * @jest-environment jsdom
 */

const mockApiRequest = jest.fn((config) => {
	//Simule o comportamento da API
	return Promise.resolve([
        {
            "mais_informacoes": "https://www.amazon.com.br/dp/B001BSX1EM?ref_=cm_sw_r_apan_dp_20ZZWBH7X76K0ENAKVDQ&language=pt-BR",
            "descricao": "O núcleo de aço de calibre pesado não deforma e aquece uniformemente sem pontos quentes para que seus biscoitos e bolos saiam do forno perfeitamente e uniformemente dourados.\nO interior de cada peça tem camadas antiaderentes interligadas que são projetadas para proporcionar liberação duradoura e de alto desempenho. Libera até 2 vezes melhor do que as assadeiras clássicas Calphalon.",
            "created": "2023-12-13T19:25:40.385Z",
            "url": "https://m.media-amazon.com/images/I/81SJ7sZHKYL._AC_UF894,1000_QL80_FMwebp_.jpg",
            "updated": "2023-12-13T19:30:02.715Z",
            "preco": "494.50",
            "nome": "Calphalon Conjunto de assadeiras antiaderentes, conjunto de 6 peças",
            "id": "27ab93168ffb6730d4a0e62336d184e9"
        },
        {
            "mais_informacoes": "https://m.magazineluiza.com.br/fritadeira-eletrica-sem-oleo-air-fryer-mondial-afo-12l-bi-oven-preta-12l-com-forno/p/236702500/ep/frel/?partner_id=64068&utm_source=pdp&utm_medium=share",
            "descricao": "A fritadeira elétrica sem óleo/Air Fryer Mondial AFO-12L-BI Oven tem as cores preto e inox e vai facilitar muito o seu dia a dia na cozinha. Fabricada em material PP e inox, tem o cesto com capacidade de 5L e oferece diversas funções como 2 em 1 onde o modelo oven alia as vantagens da tecnologia da Air Fryer ao espaço e versatilidade do forno. Tem capacidade total de 12L.",
            "created": "2023-12-13T19:21:27.170Z",
            "url": "https://a-static.mlcdn.com.br/450x450/fritadeira-eletrica-sem-oleo-air-fryer-mondial-afo-12l-bi-oven-preta-12l-com-forno/magazineluiza/236702500/b585030ba78b67a323c8eea676b1b81a.jpg",
            "updated": "2023-12-13T19:21:27.170Z",
            "preco": "724.24",
            "nome": "Fritadeira Elétrica sem óleo/Air Fryer Mondial",
            "id": "0ac9e4a0e635758cce67d5d2f6adfa7a"
        },
        {
            "mais_informacoes": "https://m.magazineluiza.com.br/kit-philco-pkt75-com-liquidificador-e-batedeira/p/235722300/ep/bate/?partner_id=64068&utm_source=pdp&utm_medium=share",
            "descricao": "Para facilitar o seu dia a dia na cozinha, você precisa ter equipamentos que auxiliem no preparo dos alimentos. Por isso você precisa conhecer o kit 101501053_110 PKT75 da Philco. Ele é composto por 1 liquidificador com 900W de potência, filtro, base antiderrapante, faca de 4 lâminas, tampa com sobre tampa, porta fio e 1 batedeira planetária com 700W de potência, 3 pares de batedores em inox para massas leves, massas pesadas e claras em neve e trabalha com 12 velocidades + Turbo.",
            "created": "2023-12-13T19:19:30.063Z",
            "url": "https://a-static.mlcdn.com.br/450x450/kit-philco-pkt75-com-liquidificador-e-batedeira/magazineluiza/235722300/218564b0a24905f772ad602c7aaed7bd.jpg",
            "updated": "2023-12-13T19:19:30.063Z",
            "preco": "344.64",
            "nome": "Kit Philco PKT75 com Liquidificador e Batedeira",
            "id": "7c50d4897b8404baf40d1ec255dbb51f"
        },
    ]);
});

describe('<Footer/>', () => {
    describe('Renderização', () => {
        test('deve renderizar Footer corretamente', async () => {
            let tree;
        
            await act(async () => {
                const clientContext = { apiRequest: mockApiRequest };

                tree = renderer.create(
                    <ClientContext.Provider value={clientContext}>
                        <Footer />
                    </ClientContext.Provider>
                );
            });
        
            expect(tree.toJSON()).toMatchSnapshot();
        });
    });

    describe('Interações', () => {
        const clientContext = { apiRequest: mockApiRequest };

        beforeEach(async () => {
            await act(async () => {
                render(<ClientContext.Provider value={clientContext}>
                    <Footer />
                </ClientContext.Provider>);
            });
        });

        it('deve testar o retorno de getPresentes', async () => {
            await waitFor(() => {
                expect(mockApiRequest).toHaveBeenCalledWith("GET", "/presentes/list");
            });
    
            await waitFor(() => {
                expect(screen.getAllByTestId('item-presente')).toHaveLength(3); 
            });
        });
    });
});