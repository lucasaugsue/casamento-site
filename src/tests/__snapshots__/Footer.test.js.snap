// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> Renderização deve renderizar Footer corretamente 1`] = `
<div>
  <div
    className="redesSociaisContainer"
  >
    <div
      className="redesSociaisText"
    >
      Siga nas redes sociais
    </div>
    <div />
    <div
      className="containerIcons"
    >
      <div
        className="iconsRedesSociais"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="InstagramIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
          />
        </svg>
      </div>
      <div
        className="iconsRedesSociais"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LinkedInIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
          />
        </svg>
      </div>
      <div
        className="iconsRedesSociais"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="GitHubIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
          />
        </svg>
      </div>
      <div
        className="iconsRedesSociais"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="GoogleIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M12.545,10.239v3.821h5.445c-0.712,2.315-2.647,3.972-5.445,3.972c-3.332,0-6.033-2.701-6.033-6.032s2.701-6.032,6.033-6.032c1.498,0,2.866,0.549,3.921,1.453l2.814-2.814C17.503,2.988,15.139,2,12.545,2C7.021,2,2.543,6.477,2.543,12s4.478,10,10.002,10c8.396,0,10.249-7.85,9.426-11.748L12.545,10.239z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="maisInformacoesContainer"
  >
    <div
      className="gridInformacoes"
    >
      <div
        className="maisInformacoesTitle"
      >
        Mais informações
      </div>
      <div
        className="maisInformacoesBody"
      >
        Para auxiliar no meu sonho em casar no meio do ano de 2024, por isso resolvi eu mesmo fazer o site do meu casamento
      </div>
    </div>
    <div
      className="gridInformacoes"
    >
      <div
        className="maisInformacoesTitle"
      >
        Presentes
      </div>
      <div
        className="textLink"
        data-testid="item-presente"
        onClick={[Function]}
      >
        Calphalon Conjunto de assadeir...
      </div>
      <div
        className="textLink"
        data-testid="item-presente"
        onClick={[Function]}
      >
        Fritadeira Elétrica sem óleo/A...
      </div>
      <div
        className="textLink"
        data-testid="item-presente"
        onClick={[Function]}
      >
        Kit Philco PKT75 com Liquidifi...
      </div>
    </div>
    <div
      className="gridInformacoes"
    >
      <div
        className="maisInformacoesTitle"
      >
        Links importantes
      </div>
      <div
        className="textLink"
        onClick={[Function]}
      >
         
        Confirme sua presença
         
      </div>
      <div
        className="textLink"
        onClick={[Function]}
      >
         
        Escreva um recado
         
      </div>
      <div
        className="textLink"
        onClick={[Function]}
      >
         
        Lista de presentes
         
      </div>
    </div>
    <div
      className="gridInformacoes"
    >
      <div
        className="maisInformacoesTitle"
      >
        Contato
      </div>
      <div
        className="boxContato"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EmailIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        <div
          className="maisInformacoesBody"
        >
          lucasaugsue7@gmail.com
        </div>
      </div>
      <div
        className="boxContato"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CallIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56-.35-.12-.74-.03-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
          />
        </svg>
        <div
          className="maisInformacoesBody"
        >
          (61)98114-6060
        </div>
      </div>
    </div>
  </div>
  <div
    className="copyrightContainer"
  >
    <div />
    <div
      className="copyrightText"
    >
      @ 2023 Copyright lucasaugsue
    </div>
    <div />
  </div>
</div>
`;
